tests/test_parameters.py

```diff
@@ -67,9 +67,6 @@ def test_being_a_variable(self, simple_linear_model):
 
         np.testing.assert_allclose(p.get_double_lower_bounds(), np.array([np.pi/2]))
         np.testing.assert_allclose(p.get_double_upper_bounds(), np.array([2*np.pi]))
-        # Test deprecated API too.
-        np.testing.assert_allclose(p.lower_bounds(), np.array([np.pi/2]))
```

Comment:
I think we should probably remove the actual API functions on the `Parameter` object too.